src/components/UpgradeCard/UpgradeCardTemplate.tsx

```
@@ -71,13 +76,15 @@ const UpgradeCardTemplate: FC<UpgradeCardTemplateProps> = ({
         </HStack>
         <Text mt={2}>{amountToConvert === "" ? "--" : formattedAmount}</Text>
         <SubmitTxButton
+          isLoading={isFetching}
           mt={10}
           onSubmit={onSubmit}
           disabled={
-            !!account &&
-            (amountToConvert == 0 ||
-              amountToConvert == "" ||
-              BigNumber.from(amountToConvert).gt(max))
+            (!!account &&
+              (amountToConvert == 0 ||
+                amountToConvert == "" ||
+                BigNumber.from(amountToConvert).gt(max))) ||
+            isBlocked
           }
         />
```

I'm wondering if instead of searching for all the buttons that are firing the transaction in the wallet we couldn't just do it at the transaction sending level.

I think in all of those buttons we use [`useSendTransactionFromFn` hook](https://github.com/threshold-network/token-dashboard/blob/main/src/web3/hooks/useSendTransaction.ts#L41), so we could probably just throw an error there before sending the transaction. The problematic thing though, is that the status could be fetching at the moment of clicking the button, so it wouldn't be the best user experience if we would throw an error when the data is still fetching (this would require the user to click the button again, which might be confusing). We could add an alert that we are checking the TRM status of the current address.

We can stick with your solution too, but there is a possibility we will miss out some of the buttons. For example I can se that we've missed the "Authorize Increase" button and some others 🤔 

Regardless, if we want to block users from doing any contract integrations in the dApp when their wallet is flagged as severe risk, we shoudl still throw an error at `useSendTransactionFromFn` hook.

---

Thanks for the review! I agree on implementing TRM validation in the useSendTransactionFromFn hook, which has been addressed in  detailed in the [add trm validation to the send transaction hook](https://github.com/threshold-network/token-dashboard/pull/736/commits/6280f15d2220718e3f1e2ea10fa7ff513633816c)

Additionally, for a better user experience, I propose we also include validation at the button level. This way, we can handle validations more effectively while maintaining user convenience. This button-level validation has been centralized in the refactored SubmitTxButton component, addressed in [add trm validation to transaction buttons](https://github.com/threshold-network/token-dashboard/pull/736/commits/9ee4918ce8f732d1849a11d8b96edf8cc83bd382) commit. This approach should ensure we don't miss any buttons and enhance overall usability. Appreciate your feedback on this combined strategy!